metrics/prometheus.go

```diff
@@ -1746,6 +1746,54 @@ func NewPrometheusCollector(i infoProvider, f ContainerLabelsFunc, includedMetri
 		})
 	}
 
+	if includedMetrics.Has(container.PressureMetrics) {
+		c.containerMetrics = append(c.containerMetrics, []containerMetric{
+			{
+				name:      "container_pressure_cpu_stalled_seconds_total",
+				help:      "Total time duration no tasks in the container could make progress due to CPU congestion.",
+				valueType: prometheus.CounterValue,
+				getValues: func(s *info.ContainerStats) metricValues {
+					return metricValues{{value: float64(s.Cpu.PSI.Full.Total) / 1000.0 / 1000.0, timestamp: s.Timestamp}}
```

Comment:
There's a helper function, `func asNanosecondsToSeconds(v uint64) float64`, let's add a similar one for this conversion.

```go
// asMicrosecondsToSeconds converts microseconds into a float64 representing seconds.
func asMicrosecondsToSeconds(v uint64) float64 {
  return float64(v) / float64(time.Microsecond)
}
```

Then we can simplify to:

```suggestion
					return metricValues{{value: asMicrosecondsToSeconds(s.Cpu.PSI.Full.Total), timestamp: s.Timestamp}}
```

Reply:
Done.

Reply:
So, I may be wrong about this. The documentation that I can find has conflicting information. In the other PR, [this comment](https://github.com/google/cadvisor/pull/3083#discussion_r1377422274) suggests that `full` is valid for cgroup CPU pressure.

I can confirm this locally:

```console
[/sys/fs/cgroup/user.slice] $ cat cpu.pressure 
some avg10=0.00 avg60=0.00 avg300=0.07 total=425967721
full avg10=0.00 avg60=0.00 avg300=0.00 total=177284511
```

I also checked the kernel docs, but it still isn't clear. :confused: